pkg/webui/console/views/gateway-add/index.js

```diff
@@ -110,6 +111,19 @@ export default class GatewayAdd extends React.Component {
 
   render () {
     const { error } = this.state
+    const { env } = this.props
+
+    const gs = env.config.gs
+
+    let gsServerAddress = ''
+    if (gs.enabled) {
+      gsServerAddress = new URL(gs.base_url).host
```

Comment:
In case of devices, the SDK takes care of setting default values. Should we follow this here too?
https://github.com/TheThingsNetwork/lorawan-stack/blob/master/sdk/js/src/service/devices/index.js#L193-L200

Reply:
it depends if we want to prefill the field in the ui

Reply:
We could do that in any case as we know what the SDK would fill the value with. As such I am ok with leaving this here. I just think it would be good to try making the SDK behavior consistent in terms of defaults values.

Reply:
Should we then have it "default" (not knowing what it becomes)?

I'm fine with both BTW, but I think it can become messy if we mix server defaults with client defaults. CC @htdvisser

Reply:
i specifically asked about this in slack. can we just agree on it once?

Reply:
The client should be responsible for this. It is impossible for the server (in this case the Identity Server) to know the address of the GS that the user wants to assign the gateway to or the NS/AS/JS that the user wants to assign the device to.

IMO it doesn't really matter if it's the SDK (JS/Go) or the client (Console/CLI) that is responsible for this. I have a slight preference for the latter (meaning that in the console it would be a pre-filled default in the form).

P.S.

Note that we're talking about the Gateway and EndDevice registrations in the Identity Server here, and not about the ApplicationLink in the Application Server.

Reply:
Just to clarify:
We were not talking about default values set in the backend here. This is about whether to set default values in the (JS) SDK and/or in the console itself. Whereas I think we should do both, because (non console) SDK users will likely want to make use of that functionality as well. In the console we can determine these default values too and use them to prefill form values, to improve clarity.
A special case would be randomly generated key values which can be generated either by the client (SDK / console) or also by the backend (to ensure validity in eg. DevAddr). Here, in any case, I think showing the value up front to the user does not add any value and can be confusing rather.

Reply:
Please create a new issue for this discussion (and mention me).

You can copy the following as my first comment on it:

---

I think we should avoid adding too much magic to the SDKs. I think defaults should either be the responsibility of the server, in which case the client doesn't know anything about them and doesn't send them to the SDK (and therefore backend) on create. The console for instance can just show a placeholder "use server default". Other defaults can be the responsibility of the client, in which case the default is shown as a placeholder, and the value is sent explicitly to the SDK on create. I think the SDKs should do exactly (and only) what the clients tell them to do, and not inject magic values.

For addresses there are multiple use cases:

- OTAA and *bring your own JoinEUI and DevEUI*
- OTAA and *bring your own DevEUI* and *request cluster JoinEUI*
- ABP and *bring your own DevAddr*
- ABP and *request cluster DevAddr*

For keys there are only two:

- bring your own keys
- securely generate keys (best to do that client side)

Reply:
⟶ https://github.com/TheThingsNetwork/lorawan-stack/issues/654